### ISSUE TITLE
Restoring spies/stubs of prototype functions of Element and "subclasses" fail

### ISSUE THREAD

**Issue by alexlawrence** [sinonjs/sinon#714]
The following code

``` javascript
var spy = sinon.spy(HTMLAnchorElement.prototype, 'click')
spy.restore()
```

fails with: `TypeError: Cannot delete property 'click' of [object Object]`

**Comment by ntkme** [comment]
@alexlawrence I found out it was a bug in browser implementation. Cloud you check #711 and see if my patch also fix this?

**Comment by alexlawrence** [comment]
Yes, it does. It seems like the code is broken. The comment state things that are not exactly true:

```
// For prototype properties try to reset by delete first.
// If this fails (ex: localStorage on mobile safari) then force a reset
// via direct assignment.
```
According to the comment you should move the direct assignment you introduced into the catch block instead of having an empty one @ntkme

**Comment by ntkme** [comment]
@alexlawrence The problem with #711 is that it does not throw an Error, so we have to check it outside of the try...catch block.

**Comment by alexlawrence** [comment]
Hmm, in my case it throws a `Type error` I thought.

**Comment by ihodes** [comment]
I believe that I'm having a similar issue, trying to stub the `confirm` function on `window`. Reverted to 1.13.0 for now fixes this for me.

**Comment by alexlawrence** [comment]
Yeah, we also pinned the version. So I guess there should be test for that? :)

**Comment by ntkme** [comment]
@alexlawrence @ihodes Have you tried v1.14.1?

**Comment by mroderick** [comment]
> I believe that I'm having a similar issue, trying to stub the confirm function on window. Reverting to 1.13.0 for now fixes this for me.

If you can reproduce this consistently, could you please open an issue for it, stating what version of Sinon.JS you're using and in what environments it fails?

**Comment by mroderick** [comment]
> This should have been fixed in v1.14.1.

Should, or is?

**Comment by ihodes** [comment]
@mroderick filed in #719 :)
